File: lib/src/main/java/com/pizzk/overlay/OverlayAdapter.java
package com.pizzk.overlay;

import com.pizzk.overlay.el.Overlay;

import java.util.Arrays;
import java.util.LinkedList;
import java.util.List;

public class OverlayAdapter {
    private final List<Overlay> overlays = new LinkedList<>();
    private OverlayLayout view = null;

    public OverlayAdapter with(OverlayLayout v) {
        view = v;
        return this;
    }

    public OverlayAdapter overlays(List<Overlay> v) {
        overlays.clear();
        if (null == v) return this;
        overlays.addAll(v);
        return this;
    }

    public OverlayAdapter overlays(Overlay... vs) {
        if (null == vs || vs.length <= 0) return this;
        return overlays(Arrays.asList(vs));
    }

    public boolean next() {
        if (null == view) return false;
        if (overlays.isEmpty()) {
            view.setOverlay(null);
            view.setVisibility(false);
            return false;
        }
        view.setVisibility(true);
        view.setOverlay(overlays.remove(0));
        return true;
    }

    public void finish() {
        overlays.clear();
        next();
    }
}

File: lib/src/main/java/com/pizzk/overlay/OverlayAdapter.kt
package com.pizzk.overlay

import com.pizzk.overlay.el.Overlay

class OverlayAdapter {
    private val overlays: MutableList<Overlay> = mutableListOf()
    private var view: OverlayLayout? = null

    fun with(v: OverlayLayout?): OverlayAdapter {
        view = v
        return this
    }

    fun overlays(v: List<Overlay>): OverlayAdapter {
        overlays.clear()
        overlays.addAll(v)
        return this
    }

    fun overlays(vararg vs: Overlay): OverlayAdapter {
        return overlays(vs.toList())
    }

    fun next(): Boolean {
        val view: OverlayLayout = view ?: return false
        if (overlays.isEmpty()) {
            view.setOverlay(null)
            return false
        }
        view.setOverlay(overlays.removeAt(0))
        return true
    }

    fun finish() {
        overlays.clear()
        next()
    }
}